src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -172,6 +179,80 @@ private PdpFeedTrailerEntry buildPdpFeedTrailerEntry(int totalDetailRecordCount,
+        if (isDetailFileLineValidCashAdvanceRequest(detailFileLine)) {
+            updateReportDataPdpFeedFileCountAndAmountTotals(reportData, totalPdpDetailRecordsCount, totalPdpDetailRecordsAmount);
+            LOG.info("updateReportDataForDetailFileLineBeingProcessed: Cash advance included in PDP feed file :  " + KFSConstants.NEWLINE + detailFileLine.toString());
+        }
+        else if (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isClonedCashAdvance()) {
+            updateReportDataClonedCashAdvanceCountAndAmountTotals(reportData, detailFileLine);
+            LOG.info("updateReportDataForDetailFileLineBeingProcessed: Cloned cash advance :  " + KFSConstants.NEWLINE + detailFileLine.toString());
+        }
+        else if (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isCashAdvanceUsedInExpenseReport()) {
+            updateReportDataCashAdvancesUsedInExpenseReportCountAndAmountTotals(reportData, detailFileLine);
+            LOG.info("updateReportDataForDetailFileLineBeingProcessed: Cash advance used in expense report :  " + KFSConstants.NEWLINE + detailFileLine.toString());
+        }
+        else if (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isDuplicatedCashAdvanceLine()) {
+            updateReportDataDuplicateCashAdvanceCountAndAmountTotals(reportData, detailFileLine);
+            updateReportDataWithFileLineValidationError(reportData, detailFileLine);
+            LOG.info("updateReportDataForDetailFileLineBeingProcessed: DUPLICATE cash advance :  " + KFSConstants.NEWLINE + detailFileLine.toString());
+        }
+        else if (detailFileLine.getValidationResult().isNotCashAdvanceLine()) {
+            updateReportDataTravelRequestsOnlyCountAndAmountTotals(reportData, detailFileLine);
+            LOG.info("updateReportDataForDetailFileLineBeingProcessed: NOT a cash advance :  " + KFSConstants.NEWLINE + detailFileLine.toString());
+        }
+        else if (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isNotValidCashAdvanceLine()) {
+            updateReportDataValidationErrorCountAndAmountTotals(reportData, detailFileLine);
+            updateReportDataWithFileLineValidationError(reportData, detailFileLine);
+            LOG.info("updateReportDataForDetailFileLineBeingProcessed: Cash Advance but validation failed :  " + KFSConstants.NEWLINE + detailFileLine.toString());
+        }
+        else {
+            LOG.info("updateReportDataForDetailFileLineBeingProcessed: ***PROBLEM*** detailFileLine Line NOT included in any report totals and this should NOT happen :  " + KFSConstants.NEWLINE + detailFileLine.toString());
```

Comment:
Changing to LOG.error. 
At this point the code could not do anything to address the issue as this would be a condition that was not considered in the spec. I don't think that we would want to fail the batch job. I will make a note in the User Story that does the emailing of the report to include a check where the sum all the "counts" and "amounts" is compared to the corresponding totals values in the attribute references on line 212 below.  If either count or amount summation of the report attributes does not match exactly to those totals then flag an alert on the email of the report being sent.